src/android/app/src/main/jni/native.cpp

```diff
@@ -74,6 +75,16 @@ namespace {
 ANativeWindow* s_surface;
 ANativeWindow* s_secondary_surface;
 
+namespace CompressionStatus {
```

Comment:
same question as in kotlin, why use jint constants instead of an enum class?

Reply:
Since you are already using a `static_cast<jint>` everywhere you use this, I think using an enum class provides additional type safety and is more modern c++, so I would go with that. The only real advantage, I think, of just using namespace constexpr like this would be not needing to cast to jint, but I don't think it matters.

I would replace this with 
```c++
enum class CompressionStatus : jint {
    Success = 0,
    Compress_Unsupported = 1,
    Compress_AlreadyCompressed = 2,
    Compress_Failed = 3,
    Decompress_Unsupported = 4,
    Decompress_NotCompressed = 5,
    Decompress_Failed = 6
};
```